programs/mango-v4/src/instructions/token_deposit.rs

```diff
@@ -189,18 +189,38 @@ impl<'a, 'info> DepositCommon<'a, 'info> {
         // Since depositing can only increase health, we can skip the usual pre-health computation.
         // Also, TokenDeposit is one of the rare instructions that is allowed even during being_liquidated.
         //
-        if !account.fixed.is_in_health_region() {
+        // TEMPORARY: (https://github.com/blockworks-foundation/mango-v4/pull/415/files)
+        // check is disabled, since we want to enforce a temporary deposit limit
+        // if !account.fixed.is_in_health_region()
+        {
             let retriever =
                 new_fixed_order_account_retriever(remaining_accounts, &account.borrow())?;
-            let health = compute_health(&account.borrow(), HealthType::Init, &retriever)
-                .context("post-deposit init health")?;
+            let cache = new_health_cache(&account.borrow(), &retriever)?;
+            let health = cache.health(HealthType::Init);
             msg!("health: {}", health);
             let was_being_liquidated = account.being_liquidated();
             let recovered = account.fixed.maybe_recover_from_being_liquidated(health);
             require!(
                 !was_being_liquidated || recovered,
                 MangoError::DepositsIntoLiquidatingMustRecover
             );
```

Comment:
Let's make this not a temporary hack. token_deposit would always build the retriever and cache, but this health check etc would be gated by `!is_in_health_region()`.